venona/cmd/start.go

```diff
@@ -133,9 +143,54 @@ func run(options startOptions) {
 	}
 	dieOnError(agent.Start())
 
-	server := server.Server{
+	serverMode := server.Release
+	if options.verbose {
+		serverMode = server.Debug
+	}
+	server, err := server.New(&server.Options{
 		Port:   fmt.Sprintf(":%s", options.serverPort),
 		Logger: log.New("module", "server"),
-	}
+		Mode:   serverMode,
+	})
+	dieOnError(err)
+
+	go handleSignals(context.Background(), &agent, &server, log)
+
 	dieOnError(server.Start())
 }
+
+func handleSignals(ctx context.Context, a *agent.Agent, s *server.Server, log logger.Logger) {
```

Comment:
do not use the server or agent here to avoid race condition, instead pass the channel

Reply:
right, will do